code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm

```diff
@@ -64,7 +64,21 @@
 		T.color = color
 
 /datum/reagent/paint/touch_obj(var/obj/O)
-	if(istype(O))
+	//special checks for special items
+	if(istype(O, /obj/item/weapon/reagent_containers))
+		return
```

Comment:
I thought it should check for containers first, and for technicolor stuff last, so I decided to separate those checks.